src/Domain/UseCase/CreateValetToken/CreateValetTokenResponse.ts

```diff
@@ -0,0 +1,15 @@
+import { Operation } from '../../Operation/Operation'
+import { ValetToken } from '../../ValetToken/ValetToken'
+
+export type ErrorResponse = {
+  message: string,
+  forbiddenOperation: Operation,
+}
+
+export type CreateValetKeyResponse = {
```

Comment:
```suggestion
export type CreateValetKeyResponse = {
  success: boolean,
  error?: {
    message: string,
    forbiddenOperation: Operation
  },
  valetToken?: ValetToken
}
```

Reply:
This is not equivalent. It allows having a `{success: true, error: something}` or `{success: false, valetToken: something}` or combinations of these. You can't take advantage of the wonderful features of TypeScript, i.e. [control flow based type analysis with type narrowing](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#control-flow-based-type-analysis).

With the current design you can't make a mistake of extracting a success value out of a erroneous response or vice versa. Same thing happens for `ValidatedValue<T>`s. With the current design you are forced to check if the value is valid or not before extracting either `error` or `value`, otherwise you'll get a type error. This is really an awesome feature, a TypeScript-specific win.

Reply:
Resolving per above.